metadata_service/proxy/neo4j_proxy.py

```diff
@@ -724,9 +725,67 @@ def get_user_detail(self, *, user_id: str) -> Union[UserEntity, None]:
                             team_name=record.get('team_name'),
                             slack_id=record.get('slack_id'),
                             employee_type=record.get('employee_type'),
+                            profile_url=record.get('profile_url'),
+                            role_name=record.get('role_name'),
+                            user_id=record.get('user_id'),
                             manager_fullname=manager_name)
         return result
 
+    @timer_with_counter
+    def put_user(self, *, user: UserEntity) -> None:
+        """
+        Update user with supplied data.
+        :param user: new user to be added
+        """
+        self.insert_user(user=user)
+
+    @timer_with_counter
+    def post_users(self, *, users: List[UserEntity]) -> None:
+        """
+        Add or update users with supplied data.
+        :param users: users to be added or updated
+        """
+
+        for user in users:
+            self.insert_user(user=user)
+
+    def insert_user(self, *, user: UserEntity) -> None:
+        """
+        Add or update user with supplied data
+        :param user: user to be added or updated
+        """
+
+        data = user.__dict__
+        data['published_tag'] = time.strftime(PUBLISH_TAG_TIME_FORMAT)
```

Comment:
Do you need `publisher_last_updated_epoch_ms` instead of `published_tag`? 
https://github.com/lyft/amundsendatabuilder/blob/master/databuilder/publisher/neo4j_csv_publisher.py#L48

In Lyft we use `published_tag` from databuilder to remove stale data via string comparison. To have timestamp, `publisher_last_updated_epoch_ms` might be better.